File: app/src/main/java/im/point/dotty/SplashActivity.java
package im.point.dotty;

import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.ViewModelProvider;

import android.os.Bundle;

import im.point.dotty.domain.AuthViewModel;
import im.point.dotty.domain.ViewModelFactory;

public class SplashActivity extends AppCompatActivity {
    private AuthViewModel viewModel;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        viewModel = new ViewModelProvider(this, new ViewModelFactory(getApplication()))
                .get(AuthViewModel.class);
    }

    @Override
    protected void onStart() {
        super.onStart();
        viewModel.resetActivityBackStack();
    }
}

File: app/src/main/java/im/point/dotty/SplashActivity.kt
/*
 * Copyright (c) 2019-2021 Pavel A. Sokolov
 */
package im.point.dotty

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.ViewModelProvider
import androidx.lifecycle.lifecycleScope
import im.point.dotty.common.ViewModelFactory
import im.point.dotty.login.LoginViewModel

class SplashActivity : AppCompatActivity() {
    private lateinit var viewModel: LoginViewModel

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        viewModel = ViewModelProvider(this, ViewModelFactory(this))
                .get(LoginViewModel::class.java)
    }

    override fun onStart() {
        super.onStart()
        lifecycleScope.launchWhenStarted { viewModel.resetActivityBackStack() }
    }
}